src/main/scala/is/hail/io/reference/FASTAReader.scala

```diff
@@ -0,0 +1,185 @@
+package is.hail.io.reference
+
+import java.io.{ObjectInputStream, ObjectOutputStream}
+import java.util
+import java.util.Map.Entry
+
+import htsjdk.samtools.reference.{FastaSequenceIndex, FastaSequenceIndexCreator, ReferenceSequenceFile, ReferenceSequenceFileFactory}
+import is.hail.HailContext
+import is.hail.utils._
+import is.hail.variant.{RGBase, ReferenceGenome, Locus}
+import org.apache.hadoop
+
+import scala.language.postfixOps
+import scala.sys.process._
+
+class SerializableReferenceSequenceFile(val hConf: SerializableHadoopConfiguration, val fastaFile: String, val indexFile: String) extends Serializable {
+  @transient var value_ : ReferenceSequenceFile = _
+
+  def value: ReferenceSequenceFile = {
+    if (value_ == null) {
+      val localFastaFile = FASTAReader.getLocalFastaFileName(hConf, fastaFile, indexFile)
+      value_ = ReferenceSequenceFileFactory.getReferenceSequenceFile(new java.io.File(localFastaFile))
+    }
+    value_
+  }
+}
+
+object FASTAReader {
+  @volatile var localFastaFiles = Map.empty[String, String]
+
+  def getLocalIndexFileName(fastaFile: String): String =
+    ReferenceSequenceFileFactory.getFastaIndexFileName(new java.io.File(fastaFile).toPath).toString
+
+  def getLocalFastaFileName(hConf: SerializableHadoopConfiguration, fastaFile: String, indexFile: String): String = {
+    synchronized {
+      if (!localFastaFiles.contains(fastaFile)) {
+        val localFile = FASTAReader.setup(hConf, fastaFile, indexFile)
+        localFastaFiles += fastaFile -> localFile
+      }
+      localFastaFiles(fastaFile)
+    }
+  }
+
+  def setup(hConf: SerializableHadoopConfiguration, fastaFile: String, indexFile: String): String = {
+    val fastaPath = new hadoop.fs.Path(fastaFile)
+    var localFastaFile = hConf.value.getTemporaryFile("file:///tmp/", suffix = Some(fastaPath.getName))
```

Comment:
Maybe I misunderstand something, but I don't see a compelling reason to allocate a `Path` and call `getName` just to get a suffix for a temporary file.